### PR TITLE
send file contents and not file name

### PR DESCRIPTION
**Pull request by swayamraina**
### What this PR does / why we need it:
implements https://github.com/apache/apisix/issues/2196

### CODE REVIEW

File: apisix/utils/log-util.lua [hunk 1]
```diff
@@ -64,7 +64,11 @@ local function get_full_log(ngx, conf)
         else
             local body_file = ngx.req.get_body_file()
             if body_file then
-                log.request.body_file = body_file
+                local lines = {}
```
Comment (membphis) [inline apisix/utils/log-util.lua:67]: https://stackoverflow.com/questions/11201262/how-to-read-data-from-a-file-in-lua

Read the file content in this way, is the right way:

```lua
local open = io.open

local function read_file(path)
    local file = open(path, "rb") -- r read mode and b binary mode
    if not file then return nil end
    local content = file:read "*a" -- *a or *all reads the whole file
    file:close()
    return content
end

local fileContent = read_file("foo.html");
print (fileContent);
```

Comment (swayamraina) [reply apisix/utils/log-util.lua:67]: `if not file then return nil end`
do we need this part of the code? if the body is not present then can we not assume the data will be present in the file?

**Comment by moonming** [comment]
ping @membphis

File: apisix/utils/log-util.lua [hunk 2]
```diff
@@ -64,7 +66,10 @@ local function get_full_log(ngx, conf)
         else
             local body_file = ngx.req.get_body_file()
             if body_file then
-                log.request.body_file = body_file
+                local file = open(body_file, "rb")
+                local content = file:read "*a"
+                file:close()
+                log.request.body_file = content
```
Comment (spacewander) [inline apisix/utils/log-util.lua:72]: We should use a new field to store the content to avoid break change.

File: apisix/utils/log-util.lua [hunk 3]
```diff
@@ -64,7 +66,10 @@ local function get_full_log(ngx, conf)
         else
             local body_file = ngx.req.get_body_file()
             if body_file then
-                log.request.body_file = body_file
+                local file = open(body_file, "rb")
```
Comment (spacewander) [inline apisix/utils/log-util.lua:69]: Need to check if `open` is successful to handle something like disk problem.

File: apisix/utils/log-util.lua [hunk 4]
```diff
@@ -64,7 +66,10 @@ local function get_full_log(ngx, conf)
...
+                local content = file:read "*a"
```
Comment (spacewander) [inline apisix/utils/log-util.lua:70]: Need to check if `read` is successful like `local content, err = file:read "*a"`

**Review by spacewander — CHANGES_REQUESTED** [review]

**Comment by Yiyiyimu** [comment]
Hi @swayamraina could you have a look at the comments

**Comment by tokers** [comment]
Better to add some cases to cover it.

**Pull request closed without merging**